cocos/tween/tween-action.ts

```diff
@@ -155,6 +155,9 @@ export class TweenAction<T extends object> extends ActionInterval {
 
             if (value.value !== undefined) {
                 customValue = value.value;
+                if (typeof customValue === 'function') {
+                    customValue = customValue();
+                }
```

Comment:
As custom progress is function, why need to support custom value as a function too? And the comment or PR title doesn't say it will support custom value as a function type.

Reply:
There are two kinds of usage of setting value, one is directly setting the value to `value` property, another is setting a callback that returns the `value`. For instance:
```ts
    // 1
    const o = new MyObject();
    tween(o)
        .to(1, { myProp: {
            value: new MyProp(100, 100), // Just set the `target value` 
            progress: MyProp.lerp,
            clone: v => v.clone(),
            legacyProgress: false,
        } })
        .start();
```

```ts
    // 2
    const o = new MyObject();
    tween(o)
        .to(1, { myProp: {
            value: () => new MyProp(100, 100), // A function returns the `target value`.
            progress: MyProp.lerp,
            clone: v => v.clone(),
            legacyProgress: false,
        } })
        .start();
```
And this is to keep the same behavior/usage as the situation in which we don't use a custom progress. E.g.

```ts
// 1
const o = new MyObject();
    tween(o)
        .to(1, { myProp:  new MyProp(100, 100) }) // Just set the `target value` 
        .start();
```

```ts
// 2
const o = new MyObject();
    tween(o)
        .to(1, { myProp:  () => new MyProp(100, 100) }) // A function returns the `target value`.
        .start();
```

Reply:
The old logic of which doesn't use a custom progress: 

![WeChat9197aa9db93c986d25be32eb01bca9ed](https://github.com/cocos/cocos-engine/assets/493372/6b67e606-24b2-4ed8-8b7d-64e01b2886be)